zio-http/src/test/scala/zio/http/URLSpec.scala

```diff
@@ -96,5 +97,40 @@ object URLSpec extends ZIOSpecDefault {
           }
         },
       ),
+      suite("queryParamsAsString")(
+        test("successfully returns the URL query params as a decoded string") {
+          val urls = Gen.fromIterable(
+            Seq(
+              "",
+              "/",
+              "/users?ord=ASC&txt=scala%20is%20awesome%21&u=1&u=2",
+              "/users",
+              "/users#the%20hash",
+              "http://abc.com",
+              "http://abc.com/",
+              "http://abc.com/list",
+              "http://abc.com/users?ord=ASC&txt=scala%20is%20awesome%21&u=1&u=2",
+              "http://abc.com/users?u=1&u=2&ord=ASC&txt=scala%20is%20awesome%21",
+              "http://abc.com/users?u=1#the%20hash",
+              "http://abc.com/users",
+              "http://abc.com/users/?u=1&u=2&ord=ASC&txt=scala%3Fis%2Bawesome%21",
+              "http://abc.com/users#the%20hash",
+              "ws://abc.com/subscriptions",
+              "ws://abc.com/subscriptions?",
+              "wss://abc.com/subscriptions",
+            ),
+          )
+
+          checkAll(urls) { genUrl =>
```

Comment:
I think this covers the added method query. 
There is another test for URL symmetry - https://github.com/zio/zio-http/blob/d9d18bb5332b3452f9e395da704e7cf0b824e077/zio-http/src/test/scala/zio/http/URLSpec.scala#L10

Reply:
Yes, also the internal implementation uses built in codecs so we are testing a library function here.
Not worth it IMO :)